### ISSUE TITLE
Feature Request: promise like play interface

### ISSUE THREAD

**Issue by devvythelopper** [goldfire/howler.js#1075]
It would be nice to use a promise like interface after a sound has ended/stopped playing. Something like

`sound.play().then(function(){ /* do something */ });`

or something similar. Suggested implementations:

	Howl.prototype.play = (function(_super) {
		return function(){
			var self = this;
			var result = new Promise(function(res){
				self.once('stop', res);
				self.once('end', res);
			});
			result.id = _super.apply(self, arguments);
			return result;
		}
	})(Howl.prototype.play);

	// or maybe
	Howl.prototype.playAnd = function(){
		var self = this;
		var result = new Promise(function(res){
			self.once('stop', res);
			self.once('end', res);
		});
		result.id = self.play.apply(self, arguments);
		return result;
	}

	// to be called like
	how.playAnd().then(function(){
		// whatever
	})

**Comment by cdcv** [comment]
+1

**Comment by suterma** [comment]
I agree that at least the fadings should have a promise-based variant. I have created promise-based fade-out handler here (only that because that is my only need). See https://github.com/suterma/replayer-pwa/blob/main/src/code/audio/HowlerFader.ts
Here is the relevant part:

    /** Returns a fade-out promise for the currently playing track
     * @remarks A fade operation is only started when the duration is non-zero
     */
    fadeOut(): Promise<void> {
        if (this.duration) {
            return new Promise((resolve, reject) => {
                try {
                    const currentVolume = this.sound.volume();
                    if (currentVolume > 0) {
                        //Determine the required fade, based on the current volume
                        //(a fade-in could be currently running, requiring a partial fade only)
                        const requiredDuration = this.duration * currentVolume;

                        this.sound.fade(currentVolume, 0, requiredDuration);

                        this.sound.once('fade', function () {
                            resolve();
                        });
                    } else {
                        resolve(); //immediately
                    }
                } catch (err) {
                    reject('Fade-out failed.');
                }
            });
        } else {
            return Promise.resolve();
        }
    }

And here's how you create it for example (after a howl was created):

                this.fader = new HowlerFader(
                    this.howl,
                    1000 /*fading duration 1 second*/
                );

Then invoke it like this:

                this.fader.fadeOut().then(() => {
                    this.sound.pause();
                });

Use this code as you like, I am too lazy to create a PR.
_License: although the referenced project is GPLv3, I hereby grant the MIT-License for the code shown here._
